Hazel/Dtls/DtlsUnityConnection.cs

```diff
@@ -213,23 +225,34 @@ protected override void ResendPacketsIfNeeded()
                     DateTime now = DateTime.UtcNow;
                     if (now >= this.nextEpoch.NextPacketResendTime)
                     {
-                        switch (this.nextEpoch.State)
+                        double negotationDuration = (now - this.nextEpoch.NegotationStartTime).TotalMilliseconds;
+                        this.nextEpoch.PacketResendCount++;
+
+                        if ((this.ResendLimit > 0 && this.nextEpoch.PacketResendCount > this.ResendLimit)
+                            || negotationDuration > this.DisconnectTimeout)
```

Comment:
Assuming DisconnectTimeout is already existing and used by consumers of the library, so I see reason to not change, but, just in case I'd recommend that the units are baked into the name, ie `DisconnectTimeoutMilliseconds` - I find this helps a lot from preventing unit mismatch mistakes

(And perhaps we can make it clearer here for updating to `negotiationDurationMilliseconds`)